src/Razor/src/Microsoft.VisualStudio.LanguageServices.Razor/FeatureFlagService.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System.ComponentModel.Composition;
+using System.Threading.Tasks;
+using Microsoft.Internal.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.Threading;
+
+namespace Microsoft.VisualStudio.Razor;
+
+[Export(typeof(IFeatureFlagService))]
+[method: ImportingConstructor]
+internal sealed class FeatureFlagService(
+    [Import(typeof(SAsyncServiceProvider))] IAsyncServiceProvider serviceProvider,
+    JoinableTaskContext joinableTaskContext)
+    : IFeatureFlagService
+{
+    private readonly JoinableTask<IVsFeatureFlags> _getVsFeatureFlagsTask = joinableTaskContext.Factory.RunAsync(
```

Comment:
This isn't a `JTF.Run`. It's `RunAsync(...)`, which is definitely allowed. In fact, that's one of the recommended ways to get asynchronous initialization out of the MEF constructor. See this [DevDiv Wiki link](https://devdiv.visualstudio.com/DevDiv/_wiki/wikis/DevDiv.wiki/24027/Threading?anchor=%E2%9D%8C-**do-not**-make-com-rpc-calls%2C-%60joinabletaskfactory.run%60-or-other-thread-switching-apis-from-mef-constructors).

Reply:
Yea, as long as `JTF.Run` is not called on a background thread it's good. The problem we've had in the past is getting services synchronously that would end up calling `JTF.Run` because how could we have known 🤷